src/main/java/ladder/domain/LadderResult.java

```diff
@@ -0,0 +1,122 @@
+package ladder.domain;
+
+import ladder.view.LadderDepthException;
+import ladder.view.PlayerException;
+import ladder.view.RewardException;
+
+import java.util.*;
+
+/**
+ * 사다리 결과 값을 저장하는 클래스
+ * <br> LadderResult radderResult = new RadderResult("pobi, crong, hi", "꽝, 1000, 2000", 5)
+ * <br> radderResult.getResultOfName("pobi")
+ * <br> radderResult.getResultOfName("all")
+ *
+ * @author heebg, hyojaekim
+ * @version 1.0 2019-05-16
+ */
+public class LadderResult {
+    public static final String LADDERRESULT_GET_RESULT_ALL = "all";
+    public static final String NOT_FIND_PLAYER = "존재하지 않는 플레이어 입니다.";
+
+    private Ladder ladder;
+    private Map<Player, Item> result;
+
+    /**
+     * 생성자
+     *
+     * @param names
+     * @param rewards
+     * @param depth
+     */
+    public LadderResult(String names, String rewards, int depth) {
+        names = PlayerException.playerNames(names);
+        rewards = RewardException.reward(rewards, names.split(",").length);
+        depth = LadderDepthException.ladderMinDepth(depth);
+        List<Player> players = getPlayers(names);
+        this.ladder = getLadder(players, depth);
+        this.result = getResult(players, convertItems(rewards));
+    }
+
+    private List<Item> convertItems(String items) {
+        List<Item> convertItems = new ArrayList<>();
+        for (String item : items.split(",")) {
+            convertItems.add(new Item(item));
+        }
+        return convertItems;
+    }
+
+    /**
+     * 이름의 결과값 반환
+     *
+     * @param name
+     * @return
+     */
+    public String getResultOfName(String name) {
+        if (name.equals(LADDERRESULT_GET_RESULT_ALL)) {
+            return getResultOfAllName();
+        }
+        return matchPlayerItem(name);
+    }
+
+    private String matchPlayerItem(String name) {
+        try {
```

Comment:
이렇게하면 더 간결하지 않을까요?
NullPointerException을 또 감쌀 필요는 없어보여요
```suggestion
        if (result == null) {
            throw new NullPointerException(NOT_FIND_PLAYER);
        }
        return result.get(new Player(name)).toString();
```

Reply:
다시 생각해보니 null 값이라면 다른 곳에서 예외 처리를 하고 있고 
결과가 알고 싶은 플레이어가 있는지 없는지 체크해야 하기 때문에 메소드를 분리해서 예외를 던지도록 처리하였습니다!